From 1b72897d7dfd791c2ce89946b6cfd1b896e36517 Mon Sep 17 00:00:00 2001
From: Neville Li <neville@spotify.com>
Date: Thu, 29 Jan 2015 17:31:04 -0800
Subject: [PATCH 66/93] PARQUET-142: add path filter in ParquetReader
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Currently parquet-tools command fails when input is a directory with _SUCCESS file from mapreduce. Filtering those out like ParquetFileReader does fixes the problem.

```
parquet-cat /tmp/parquet_write_test
Could not read footer: java.lang.RuntimeException: file:/tmp/parquet_write_test/_SUCCESS is not a Parquet file (too small)

$ tree /tmp/parquet_write_test
/tmp/parquet_write_test
├── part-m-00000.parquet
└── _SUCCESS
```

Author: Neville Li <neville@spotify.com>

Closes #89 from nevillelyh/gh/path-filter and squashes the following commits:

7377a20 [Neville Li] PARQUET-142: add path filter in ParquetReader
---
 .../main/java/parquet/hadoop/ParquetReader.java    |    8 +++++++-
 1 files changed, 7 insertions(+), 1 deletions(-)

diff --git a/parquet-hadoop/src/main/java/parquet/hadoop/ParquetReader.java b/parquet-hadoop/src/main/java/parquet/hadoop/ParquetReader.java
index ec839e2..4d80f0f 100644
--- a/parquet-hadoop/src/main/java/parquet/hadoop/ParquetReader.java
+++ b/parquet-hadoop/src/main/java/parquet/hadoop/ParquetReader.java
@@ -30,6 +30,7 @@ import org.apache.hadoop.fs.FileStatus;
 import org.apache.hadoop.fs.FileSystem;
 import org.apache.hadoop.fs.Path;
 
+import org.apache.hadoop.fs.PathFilter;
 import parquet.filter.UnboundRecordFilter;
 import parquet.filter2.compat.FilterCompat;
 import parquet.filter2.compat.FilterCompat.Filter;
@@ -113,7 +114,12 @@ public class ParquetReader<T> implements Closeable {
     this.conf = conf;
 
     FileSystem fs = file.getFileSystem(conf);
-    List<FileStatus> statuses = Arrays.asList(fs.listStatus(file));
+    List<FileStatus> statuses = Arrays.asList(fs.listStatus(file, new PathFilter() {
+      @Override
+      public boolean accept(Path p) {
+        return !p.getName().startsWith("_") && !p.getName().startsWith(".");
+      }
+    }));
     List<Footer> footers = ParquetFileReader.readAllFootersInParallelUsingSummaryFiles(conf, statuses, false);
     this.footersIterator = footers.iterator();
     globalMetaData = ParquetFileWriter.getGlobalMetaData(footers);
-- 
1.7.0.4

